framework/bundles/org.jboss.tools.rsp.launching.java/src/main/java/org/jboss/tools/rsp/launching/java/ArgsUtil.java

```diff
@@ -0,0 +1,238 @@
+		if( haystack == null )
+			return false;
+		return Arrays.asList(haystack).contains(needle);
+	}
+	
+	public static boolean matchesLongArg(String needle, String[] haystack) {
+		if( haystack == null )
+			return false;
+		for( int i = 0; i < haystack.length; i++ ) {
+			if( needle.startsWith(haystack[i] + EQ) || needle.startsWith(QUOTE + haystack[i] + EQ))
+				return true;
+		}
+		return false;
+	}
+	
+	public static String setArg(String allArgs, String shortOpt, String longOpt, String value ) {
+		if( value != null && value.contains(SPACE)) {
+			boolean startsWith = value.startsWith(QUOTE);
+			boolean endsWith = value.endsWith(QUOTE);
+			boolean and = startsWith && endsWith;
+			if( !and )
+				value = QUOTE + value + QUOTE;
+		}
+		return setArg(allArgs, shortOpt, longOpt, value, false);
+	}
+	
+	public static String setArg(String allArgs, String shortOpt, String longOpt, String value, boolean addQuotes ) {
+		String[] shortOpt2 = shortOpt == null ? new String[0] : new String[]{shortOpt};
+		String[] longOpt2 = longOpt == null ? new String[0] : new String[]{longOpt};
+		return setArg(allArgs, shortOpt2, longOpt2, value, addQuotes);
+	}
+	
+	/**
+	 * Replace (or add) an argument. 
+	 * Parse through the "allArgs" parameter to create a list of arguments.
+	 * Compare each element in allArgs until you find a match against 
+	 * one of the short argument (-b value) or long argument (--host=etcetcetc)
+	 * patterns. The set of short and long form arguments should be 100% interchangeable,
+	 * and the caller must not have a preference which is ultimately returned. 
+	 * 
+	 * If a match is found, and the match is in the short-form arguments, 
+	 * do not change the arg (-b value), but update the value in the next segment. 
+	 * 
+	 * If a match is found and it is a long form argument, replace the string
+	 * (ex:  --host=localhost) with the first longOpt (--newLongOpt=127.0.0.1)
+	 * 
+	 * @param allArgs
+	 * @param shortOpt
+	 * @param longOpt An array of possible long-form options
+	 * @param value The new value, or null if you want to clear the option
+	 * @param addQuotes
+	 * @return
+	 */
+	public static String setArg(String allArgs, String[] shortOpt, String[] longOpt, String value, boolean addQuotes ) {
+		String rawValue = (value != null && addQuotes) ? value : getRawValue(value);
+		value = (value != null && addQuotes) ? (QUOTE + value + QUOTE) : value;
+
+		boolean found = false;
+		String[] args = parse(allArgs);
+		
+		for( int i = 0; i < args.length && !found; i++ ) {
+			if( matchesShortArg(args[i], shortOpt)) {
+				if( value == null ) {
+					args[i] = null;
+				}
+				args[i+1] = value;
+				found = true;
+			} else if( matchesLongArg(args[i], longOpt)) {
+				if( rawValue == null ) {
+					args[i] = null;
+				} else {
+					String newVal = args[i].startsWith(QUOTE) ? 
+							(QUOTE + longOpt[0] + EQ + rawValue + QUOTE) 
+							: longOpt[0] + EQ + value;
+					args[i] = newVal;
+				}
+				found = true;
```

Comment:
l168-176 could be extracted to a method #processLongArg which returns ```true``` to get a smaller #setArg method which gets easier to read.

Reply:
I (personally) do not like extracting methods that change the values of an array parameter. I realize java allows this, and changing the values in the array will be maintained after the return call, but i find it extremely hard to read and follow when I do this... especially since I'm iterating over the array at the same time.